### PR TITLE
wire up feature-gate for component slis

### PR DESCRIPTION
**Pull request by logicalhan**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Puts `/metrics/slis` behind a feature gate. 

#### Does this PR introduce a user-facing change?
```release-note
Introduce `ComponentSLIs` alpha feature-gate for component SLIs metrics endpoint.
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/config.go [hunk 1]
```diff
@@ -318,6 +319,10 @@ type AuthorizationInfo struct {
 	Authorizer authorizer.Authorizer
 }
 
+func init() {
+	features.AddFeatureGates(utilfeature.DefaultMutableFeatureGate)
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/server/config.go:323]: other call sites seem to wrap in a utilruntime.Must?

Comment (logicalhan) [reply staging/src/k8s.io/apiserver/pkg/server/config.go:323]: ah, done.

**Pull request merged**